src/fetchers/PricesService.ts

```diff
@@ -0,0 +1,158 @@
+import {Consola} from "consola";
+import {timeout} from "promise-timeout";
+import fetchers from "./index";
+import ManifestHelper, {TokensBySource} from "../manifest/ManifestParser";
+import {
+  Aggregator,
+  Credentials,
+  Manifest,
+  PriceDataAfterAggregation,
+  PriceDataBeforeAggregation,
+  PriceDataFetched
+} from "../types";
+import {trackEnd, trackStart} from "../utils/performance-tracker";
+import {v4 as uuidv4} from 'uuid'
+import ManifestConfigError from "../manifest/ManifestConfigError";
+
+const logger = require("../utils/logger")("PricesFetcher") as Consola;
+
+export type PricesDataFetched = { [source: string]: PriceDataFetched[] };
+export type PricesBeforeAggregation = { [token: string]: PriceDataBeforeAggregation }
+
+export default class PricesService {
+
+  constructor(
+    private manifest: Manifest,
+    private credentials: Credentials
```

Comment:
so how exactly do you propose to pass them to fetchers? fetchers are being called in PriceService (and nowhere else).
TBH, if it would entirely depend on me, each featcher would be a "normal" class implementing Fetcher interface + I would use some IoC mechanism (like inversify) to inject requried dependencies into them. I believe this would also make testing and mocking much easier (we could have "prod" context, with real implementations, and test context, with mocks).

Reply:
I though about getting credentials directly from fetchers. Price service should not be aware of the fact that some fetchers may need some credentials.

Reply:
so fetchers would have to read config file directly? not sure, if I like this idea (as this would very tightly couple fetchers with the details of Node configuration).
We'll talk about this (note, that Fetcher interface has "opts" parameter (with "credentials" inside) from the very beginning)...

Reply:
It's not the big deal. I'm just not convinced that PriceService should be aware of the fact that some fetchers may require credentials. It seems to me that it's the responsibility of a fetcher to get the credentials if necessary.